net.sf.eclipsecs.core/src/net/sf/eclipsecs/core/config/meta/MetadataFactory.java

```diff
@@ -155,6 +220,176 @@ public static RuleMetadata getRuleMetadata(String name) {
     return metadata;
   }
 
+  /**
+   * Create metadata for modules not present in the previously eclipse provided metadata.
+   * Work in progress.
+   *
+   * @param moduleDetails module details fetched from checkstyle metadata
+   * @return ruleMetadata for the module
+   */
+  public static RuleMetadata createRuleMetadata(ModuleDetails moduleDetails) {
+    final String[] packageTokens = moduleDetails.getFullQualifiedName().split("\\.");
+    RuleGroupMetadata group;
+    String modulePackageNameString = moduleDetails.getFullQualifiedName();
+    // standard checkstyle module
+    if (modulePackageNameString.startsWith("com.puppycrawl.tools.checkstyle")) {
+      group = getRuleGroupMetadata(sPackageToGroupNameMap
+              .get(packageTokens[packageTokens.length - 2]));
+    }
+    // third party extension modules
+    else {
+      String lookupKey = findLookupKey(modulePackageNameString);
+      String ruleGroupNameString = sThirdPartyRuleGroupMap.get(lookupKey).get("name");
+      group = getRuleGroupMetadata(ruleGroupNameString);
+      // if the group of the new check hasn't been formed yet
+      // and put into the sRuleGroupMetadata map
+      if (group == null) {
+        final String ruleDescription = sThirdPartyRuleGroupMap.get(lookupKey).get("description");
+        final int rulePriority = Integer.parseInt(sThirdPartyRuleGroupMap.get(lookupKey)
+                .get("priority"));
+        group = new RuleGroupMetadata(ruleGroupNameString, ruleDescription, false, rulePriority);
+        sRuleGroupMetadata.put(ruleGroupNameString, group);
+      }
+    }
+
+    RuleMetadata ruleMeta = new RuleMetadata(moduleDetails.getName(), moduleDetails.getName(),
+            moduleDetails.getParent(), MetadataFactory.getDefaultSeverity(),
+            false, true, true, false, group);
+    ruleMeta.setDescription(moduleDetails.getDescription());
+    moduleDetails.getProperties().forEach(modulePropertyDetails -> ruleMeta.getPropertyMetadata()
+            .add(createPropertyConfig(moduleDetails, modulePropertyDetails)));
+
+    return ruleMeta;
+  }
+
+  /**
+   * Generate all prefix strings from the packagName and find which is a valid key in
+   * {@code sThirdPartyRuleGroupMap}.
+   */
+  private static String findLookupKey(String packageName) {
+    final String[] packageTokens = packageName.split("\\.");
+    List<String> prefixStringList = new ArrayList<>();
```

Comment:
`prefixStringList`, 
please do not use `String` in name, it is better to focus on content of variable, rather then type.

Reply:
Done